activestorage/app/models/active_storage/attachment.rb

```diff
@@ -15,17 +15,18 @@ class ActiveStorage::Attachment < ActiveRecord::Base
   delegate_missing_to :blob
 
   after_create_commit :analyze_blob_later, :identify_blob
+  after_destroy_commit :purge_dependent_blob_later
```

Comment:
We're switching to `delete` below because of this, right? Is this callback just to ensure that we always honor the `dependent: :destroy` from the macros?

Reply:
Yes. This, along with the switch to `dependent: :destroy` on attachment associations, keeps changes from having to track removed attachments and purge them on commit: they can just replace the associated attachments and let AR destroy the removed ones.